pkg/dockerregistry/server/auth.go

```diff
@@ -310,13 +333,19 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 					return nil, ac.wrapErr(ctx, ErrUnsupportedAction)
 				}
 				if err := verifyCatalogAccess(ctx, osClient); err != nil {
+					if len(bearerToken) == 0 && err == ErrOpenShiftAccessDenied {
+						return nil, ac.wrapErr(ctx, ErrTokenRequired)
+					}
 					return nil, ac.wrapErr(ctx, err)
 				}
 			default:
 				return nil, ac.wrapErr(ctx, ErrUnsupportedResource)
```

Comment:
why doesn't this one have a bearer token length check and return ErrTokenRequired?

Reply:
If you made a request without a token or with an anonymous token and got EPERM, then you need to get the correct token. Otherwise, it is a real EPERM.

Reply:
@bparees I do not return EPERM here only if there is no bearer. I redirect the user for the token. The token endpoint will [create a bearer token](https://github.com/openshift/image-registry/blob/master/pkg/dockerregistry/server/token.go#L54) even if you did not have it and next time I will return EPERM.

Reply:
ErrUnsupportedResource doesn't look like a redirect to get a token?

Reply:
If you talking about `default` case, then this is a handling of the situation when the user came with the wrong scope. This is not the absence of a token but an error.

Reply:
I did not immediately understand that you are not asking about the code that I changed, but about the code from the context.